common/arg.cpp

```diff
@@ -220,12 +226,25 @@ struct curl_slist_ptr {
 #define CURL_MAX_RETRY 3
 #define CURL_RETRY_DELAY_SECONDS 2
 
-static bool curl_perform_with_retry(const std::string & url, CURL * curl, int max_attempts, int retry_delay_seconds, const char * method_name) {
+static bool curl_perform_with_retry(const std::string & url,
+                                    CURL *              curl,
+                                    int                 max_attempts,
+                                    int                 retry_delay_seconds,
+                                    const char *        method_name,
+                                    const std::string & path_temporary = "") {
     int remaining_attempts = max_attempts;
 
     while (remaining_attempts > 0) {
         LOG_INF("%s: %s %s (attempt %d of %d)...\n", __func__ , method_name, url.c_str(), max_attempts - remaining_attempts + 1, max_attempts);
 
+        if (std::filesystem::exists(path_temporary)) {
```

Comment:
@ngxson I suspect you could have written a lot of this. This could be a follow on PR. But is there an argument to be made that we should check things like "Last-Modified" header before each retry? For the rare case that the model got updated in between requests.